### DISCUSSION TITLE
Adding job with file buffer as data

### DISCUSSION THREAD

**Discussion by Omers-Frontegg** [taskforcesh/bullmq#2419] [Q&A]
Say i have a service that accepts uploaded files and a worker service that processes those files.
Is it a bad practice to pass the files buffer as jobs data?

**Comment by manast** [comment]
Yes, it is a bad practice, as Redis RAM is usually quite scarce in relation to disk. So normally you will just store the file elsewhere (S3 or other similar file storage) and just store in the jobs the links to the data.

**Reply by Rush** [reply]
For a send-email queue, my attachment is <4 KB. Is there any reason to base64 it, or can I somehow store a raw Buffer in the BullMQ job? (S3 feels wasteful for something this small.)

**Reply by manast** [reply]
if it is so small then it should be ok.
